src/python/pants/init/extension_loader.py

```diff
@@ -113,6 +114,8 @@ def load_build_configuration_from_source(
     for backend_package in backend_packages:
         load_backend(build_configuration, backend_package)
 
+    build_configuration.register_rules(query_rules())
```

Comment:
More idiomatic is registering here:

https://github.com/pantsbuild/pants/blob/22cc05f53fe5e4619e42f6ada5d13efef1a2ec59/src/python/pants/init/engine_initializer.py#L289-L307

Reply:
Yes, these need to be there to work unfortunately. I forget the exact error when it's removed.

Reply:
Added a lengthy TODO describing why this needs to go here.

Reply:
Thank you for the explanation! I see. Hm. Agreed that's fine to be a TODO. Thanks.